src/main/java/com/thomashan/coup/turn/state/WaitingRevealCardState.java

```diff
@@ -43,12 +47,17 @@ private WaitingRevealCardState(Players players, Player player, List<Action> acti
         }
     }
 
-    static WaitingRevealCardState of(Players players, Player player, List<Action> actionHistory, Player revealer) {
-        return new WaitingRevealCardState(players, player, actionHistory, revealer);
+    static WaitingRevealCardState of(Players players, Player player, MainAction mainAction, List<Action> actionHistory, Player revealer) {
+        return new WaitingRevealCardState(players, player, mainAction, actionHistory, revealer);
     }
 
-    static WaitingRevealCardState of(Players players, Player player, List<Action> actionHistory, Player revealer, Player target) {
-        return new WaitingRevealCardState(players, player, actionHistory, revealer, target);
+    static WaitingRevealCardState of(Players players, Player player, MainAction mainAction, List<Action> actionHistory, Player revealer, Player target) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/thomashan/coup-java/pullRequest?prid=4561385)